src/main/java/org/opensearch/knn/training/TrainingJobRunner.java

```diff
@@ -176,4 +193,122 @@ private void serializeModel(TrainingJob trainingJob, ActionListener<IndexRespons
     public int getJobCount() {
         return jobCount.get();
     }
+
+    /**
+     * This method is called whenever the cluster state changes. It is used to update models that are still training when a node leaves or the cluster crashes.
+     * @param event the event that changed the cluster change
+     */
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        if (event.localNodeClusterManager()) {
```

Comment:
Could you help me to understand what does this condition check?

Reply:
It checks whether the local node where `clusterChanged()` is called is the cluster manager.  This just ensures that only the cluster manager handles updating the models

Reply:
Mentioned in other place but it would be better to add to listener only when the node is cluster manager node then.